File: src/main/java/hilfling/backend/hilfling/controllers/ArticleController.java
package hilfling.backend.hilfling.controllers;

import hilfling.backend.hilfling.model.Article;
import hilfling.backend.hilfling.service.ArticleService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/api/v1/articles")
public class ArticleController extends GenericBaseControllerImplementation<Article> {
    @Autowired
    private ArticleService service;
    public ArticleService getService() {
        return service;
    }
}

File: src/main/kotlin/hilfling/backend/hilfling/controller/ArticleController.kt
package hilfling.backend.hilfling.controller

import hilfling.backend.hilfling.model.Album
import hilfling.backend.hilfling.model.Article
import hilfling.backend.hilfling.model.PhotoGangBanger
import hilfling.backend.hilfling.model.SecurityLevel
import hilfling.backend.hilfling.repository.AlbumRepository
import hilfling.backend.hilfling.repository.ArticleRepository
import me.liuwj.ktorm.database.Database
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/article")
class ArticleController {
    val repository = ArticleRepository()

    @GetMapping("/{id}")
    fun getById(@PathVariable("id") id: Long): Article? {
        return repository.findById(id)
    }

    @GetMapping
    fun getAll(): List<Article> {
        return repository.findAll()
    }

    @PostMapping
    fun create(
            @RequestParam("title") title: String,
            @RequestParam("plainTextBody") plainTextBody: String,
            @RequestParam("securityLevel") securityLevel: SecurityLevel,
            @RequestParam("photoGangBanger") photoGangBanger: PhotoGangBanger
    ): Article {
        return repository.create(
                title,
                plainTextBody,
                securityLevel,
                photoGangBanger
        )
    }
}